src/Form.jsx

```diff
@@ -161,12 +161,12 @@ export default class Form extends React.Component {
       propsPatch: {
         value: nextValue
       }
-    }).then(() => {
+    }).then(({ nextProps }) => {
       const { onChange } = fieldProps;
 
       if (onChange) {
         /* Call client-side onChange handler function */
-        onChange(event, nextValue, fieldProps, this.props);
+        onChange(event, nextValue, nextProps, this.props);
```

Comment:
* Please pass it as an Object: `{ event, nextValue, fieldProps, formProps: this.props }`

Note that `nextProps` are considered `fieldProps` at the moment of change. I think there's no need to class them `next`.